src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -1122,6 +1210,53 @@ public static void Xml_BaseClassAndDerivedClass2WithSameProperty()
         Assert.StrictEqual(value.IntProperty, actual.IntProperty);
         Assert.Equal(value.StringProperty, actual.StringProperty);
         Assert.Equal(value.ListProperty.ToArray(), actual.ListProperty.ToArray());
+
+        // All base properties have been hidden, so they should be default here in the base class
+        BaseClassWithSamePropertyName castAsBase = (BaseClassWithSamePropertyName)actual;
+        Assert.StrictEqual(default(DateTime), castAsBase.DateTimeProperty);
+        Assert.StrictEqual(0, castAsBase.IntProperty);
+        Assert.Null(castAsBase.StringProperty);
```

Comment:
Just a style thing, but I would have used default(int) to express that you are expecting the default value rather than I'm looking for the number 0 which just happens to be the same as the default value for int. I know it's logically the same, just the way the code is expressed. But that's just a personal style preference.